new-post: reana-0.8.1 (#21)
reanahub/blog.reana.io: content/posts/2022/reana-0.8.1.md

---
title: "REANA 0.8.1 is released"
date: 2022-02-15T09:00:00+01:00
tags:
  - release
  - client
  - cluster
---

We are glad to announce the release of REANA 0.8.1, a minor update that allows
users to set job timeout limits, improves workspace web page file name
filtering, allows displaying workspace HTML reports directly, and more!

<!--more-->

## What's new for the users?

### Configurable job timeout limits

If your workflow launches many hundreds of jobs and some of these may be

> "running away" --- for example a job enters a state of infinite loop waiting

Author: ```suggestion
"running away" --- for example, a job enters a state of infinite loop waiting
```

Reviewer: I would like to avoid using comma inside triple-dashes, so changed to "for example when a job enters..."

``kubernetes_job_timeout`` clause in the workflow specification:

```yaml {hl_lines=[5]}
  steps:
    - name: mystep
      environment: 'mydockerimage:1.0.0'
      compute_backend: kubernetes
      kubernetes_job_timeout: 3600
      commands:
        - ./mycommand mydata.root
```

You can set your preferred job timeout values (in seconds) differently in each
workflow step. If a job exceeds the timeout, it will be forced to terminate. In
this way your workflows will not get "stuck" and you will be able to inspect
workflow logs for possible root cause causing the jobs to exceed the time
limits.

If you don't specify any time limit, note that the default time limit for any
job will be _7 days_.

Please see the [custom job
timeouts](https://docs.reana.io/advanced-usage/compute-backends/kubernetes/#custom-job-timeouts)
documentation page to know more about how to set the job timeout values in your
CWL, Serial, Snakemake, or Yadage workflows.

### Searching workspace files by name

If your workflow generates hundreds of files in the workspace, it is not easy
to locate some particular file on the workspace tab of the workflow web page.

> REANA 0.8.1 now allows to search your workspaces for desired file names. Inside

Author: ```suggestion
REANA 0.8.1 now allows searching your workspaces for desired file names. Inside
```

## What's new for the administrators?

### Configure maximum job timeout limits

The newly introduced user-configurable job timeout limit feature forces
workflows to terminate when a certain amount of time is exceeded. In addition,
the cluster administrators can configure the maximum timeout limit for all user
jobs globally. This feature is always enabled and, by default, it has a value
of _7 days_ (604800 seconds). This means that, if a workflow job keeps running
for more than 7 days, it will be automatically terminated. This global
catch-all value can be configured by cluster administrators by setting the
``kubernetes_jobs_timeout_limit`` [Helm
value](https://github.com/reanahub/reana/blob/0.8.1/helm/reana/README.md).

As an administrator, you can also set the maximum custom job timeout limit that
the users will be allowed to set in their workflow jobs. By default this value
is _14 days_ (1209600 seconds) and it can be customised at the deployment time
via the ``kubernetes_jobs_max_user_timeout_limit`` [Helm
value](https://github.com/reanahub/reana/blob/0.8.1/helm/reana/README.md).

### Declare supported compute backends

> As of REANA 0.8.1 you can declare which of the compute backends (HTCondor,

Author: ```suggestion
As of REANA 0.8.1, you can declare which of the compute backends (HTCondor,
```

Kubernetes, Slurm) are supported in your REANA deployment. The users will be
able to see this information using the `info` command:

```
$ reana-client info
List of supported compute backends: kubernetes
```

This is useful to know which kind of workflows a user can launch on the given
REANA cluster instance the user is connected to. The information is also used
by `reana-client validate` to detect incompatibilities before the workflow
submission.

You can configure which compute backends are supported in your REANA deployment
by setting the ``compute_backends`` [Helm
value](https://github.com/reanahub/reana/blob/0.8.1/helm/reana/README.md).

### Configure fine-grained server logging

The uWSGI configuration of the REANA-Server component has been changed to log

> all HTTP requests to by default. In addition, you can also configure the

Author: ```suggestion
all HTTP requests by default. In addition, you can also configure the
```
